cou/steps/execute.py

```diff
@@ -54,29 +55,27 @@ def normal(text: str) -> str:
         :return: text formatted.
         :rtype: str
         """
-        return Style.RESET_ALL + Fore.RED + text + Style.RESET_ALL
+        return Style.RESET_ALL + text + Style.RESET_ALL
 
-    return (
-        normal(parameter + " (")
-        + bold("c")
-        + normal(")ontinue/(")
-        + bold("a")
-        + normal(")bort/(")
-        + bold("s")
-        + normal(")kip:")
-    )
+    return normal(parameter + "Continue(") + bold("y") + normal("/") + bold("N") + normal("):")
 
 
-async def _run_step(step: UpgradeStep, interactive: bool) -> None:
+async def _run_step(step: BaseStep, interactive: bool, indicate_progress: bool) -> None:
     """Run step and all its sub-steps.
 
     :param step: Plan to be executed on steps.
-    :type step: UpgradeStep
+    :type step: BaseStep
     :param interactive: Whether to run upgrade step in interactive mode.
     :type interactive: bool
+    :param indicate_progress: Whether to enable progress indicator for running the step.
+    :type indicate_progress: bool
     """
-    logger.debug("running step %s", step)
-    await step.run()
+    if indicate_progress:
```

Comment:
I think that progress indicator should be done only for `UpgradeStep`, so there is no need to have `indicate_progress`.

Reply:
That's true. The initial idea for reworking steps was different so I implemented a generic solution for progress indicator. After merging #163, I will update this accordingly.